src/lib/crypto/keystore/base.ts

```diff
@@ -39,6 +39,16 @@ export default class KeyStoreBase {
     return checkIsKeyPair(maybeKey)
   }
 
+  async fingerprintPublicKey(): Promise<string> {
+    const keyPair = await this.keyPair()
+    const bytes = await common.exportKeyBytes(keyPair.publicKey as PublicKey, ExportKeyFormat.SPKI)
```

Comment:
We don't want to be using SPKI but rather the PKCS8 format (which is basically just PEM). SPKI is for full certificate information which we don't need. They both still get represented as DER under the hood but PKCS#8 is just the key information, it doesn't include additional certificate information.